scripts/kind-e2e/e2e.sh

```diff
@@ -1,36 +1,56 @@
+            echo Cluster clister${i} already exists, skipping cluster creation...
+        else
+            if [[ -n "$2" ]]; then
+                kind create cluster --image=kindest/node:v$2 --name=cluster${i} --wait=5m --config=./kind-e2e/cluster${i}-config.yaml
+            else
+                kind create cluster --name=cluster${i} --wait=5m --config=./kind-e2e/cluster${i}-config.yaml
+            fi
+            master_ip=$(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' cluster${i}-control-plane | head -n 1)
+            sed -i -- "s/user: kubernetes-admin/user: cluster$i/g" $(kind get kubeconfig-path --name="cluster$i")
+            sed -i -- "s/name: kubernetes-admin.*/name: cluster$i/g" $(kind get kubeconfig-path --name="cluster$i")
+
+            if [[ $1 = keep ]]; then
+                cp -r $(kind get kubeconfig-path --name="cluster$i") ../output/kind-config/local-dev/kind-config-cluster${i}
+            fi
+
+            sed -i -- "s/server: .*/server: https:\/\/$master_ip:6443/g" $(kind get kubeconfig-path --name="cluster$i")
+            cp -r $(kind get kubeconfig-path --name="cluster$i") ../output/kind-config/dapper/kind-config-cluster${i}
+            export KUBECONFIG=$(kind get kubeconfig-path --name=cluster1):$(kind get kubeconfig-path --name=cluster2):$(kind get kubeconfig-path --name=cluster3)
+        fi
     done
-
-    export KUBECONFIG=$(kind get kubeconfig-path --name=cluster1):$(kind get kubeconfig-path --name=cluster2):$(kind get kubeconfig-path --name=cluster3)
 }
 
 function install_helm() {
+    helm init --client-only
+    helm repo add submariner-latest https://releases.rancher.com/submariner-charts/latest
     for i in 1 2 3; do
         kubectl config use-context cluster${i}
-        kubectl -n kube-system create serviceaccount tiller
-        kubectl create clusterrolebinding tiller --clusterrole=cluster-admin --serviceaccount=kube-system:tiller
-        helm init --service-account tiller
-        kubectl -n kube-system  rollout status deploy/tiller-deploy
+        if kubectl -n kube-system rollout status deploy/tiller-deploy > /dev/null 2>&1; then
+            echo Helm already installed on clister${i}, skipping helm installation...
+        else
+            echo Installing helm on clister${i}.
+            kubectl -n kube-system create serviceaccount tiller
+            kubectl create clusterrolebinding tiller --clusterrole=cluster-admin --serviceaccount=kube-system:tiller
+            helm init --service-account tiller
+            kubectl -n kube-system  rollout status deploy/tiller-deploy
+        fi
     done
 }
 
 function setup_broker() {
     kubectl config use-context cluster1
-    helm install submariner-latest/submariner-k8s-broker \
-         --name ${SUBMARINER_BROKER_NS} \
-         --namespace ${SUBMARINER_BROKER_NS}
+    if kubectl get crd clusters.submariner.io > /dev/null 2>&1; then
```

Comment:
Yes totally make sense , great suggestion i will test this.

Reply:
> we will need to figure out at some point if the CRDs need to be updated.
> 
> Could it make sense to remove-create always?

If we need to update the CRDs, we presumably need to upgrade Submariner, don’t we? I was thinking about this too when reading through the code, and my thought then was that we should KISS for now — it’s easy enough to delete everything and re-deploy...

Reply:
This will require a larger refactor. The CRDs are created with helm if we remove the CRDs and helm resources we will have to re install subm on all the gateway nodes, because the endpoints are inserted during subm helm installation on  gateway nodes, lets keep it the way it is now and add this feature later.

Reply:
> Lets keep it the current way, and add this feature later?

Yes, that’s what I’m thinking too.

Reply:
I'm fine with that, ack :) thanks for the explanation.